### DISCUSSION TITLE
data volume permissions

### DISCUSSION THREAD

**Discussion by fsaintma** [JanusGraph/janusgraph-docker#136] [Q&A] [answered]
Hi

When I launch your Docker image with a volume to map the JanusGraph data directory on my computer (the directory on my computer don't exist yet) I obtain this result and the storage is not initialized correctly

``` 
$ docker run --rm --name janusgraph-default -v ${PWD}/graph:/var/lib/janusgraph docker.io/janusgraph/janusgraph:latest
chown: changing ownership of '/var/lib/janusgraph': Operation not permitted
chmod: changing permissions of '/var/lib/janusgraph': Operation not permitted
waiting for storage...
waiting for storage...
waiting for storage...

$ ls -la
drwxr-xr-x  2 root     root      4096 mai   27 12:06 graph
``` 
The only workaround I found is to create/modify manually the directory before the Docker command with the good permissions (uid/gid = 999). 
Could we have a directory initialized with the good permissions ?
Thanks for your help.
Best regards.

**Comment by FlorianHockmann** [comment]
What you describe sounds exactly like what a user described here on Stack Overflow in general for mounting a volume into a docker container where the host path doesn't exist yet: https://stackoverflow.com/questions/44574078/mount-non-existing-host-directory-into-non-root-container

The JanusGraph container runs as user `janusgraph` by default so the situation should be exactly as described in the SO post. So, the options suggested [in the answer](https://stackoverflow.com/a/44574425/6753576) should also work with the JanusGraph container.

**Comment by fsaintma** [comment, marked as answer]
Hi thanks for your anwser. 
I have just found a workaround for my need. I write my own Dockerfile which reference your image and install [gosu](https://github.com/tianon/gosu). Finally in the docker-entrypoint.sh I replace your last line `exec "$@"`by `exec /usr/local/bin/gosu janusgraph "$@"`. It works well and it creates a directory in case it doesn't exists with the right permissions.
Hope that it helps other users.
Thanks for your job and the janusgraph-docker project.
Best regards.
